addons/account/static/tests/x2many_buttons_tests.js

```diff
@@ -0,0 +1,155 @@
+                        { id: 6, display_name: "Bill 6", duplicated_ref_ids: [1,2,3,4,5] , ref: "b6"},
+                    ],
+                }
+            }
+        };
+        setupViewRegistries();
+    });
+
+    QUnit.module("X2ManyButtonsField");
+
+    QUnit.test("component rendering: less than 3 records on field", async function (assert) {
+        assert.expect(2);
+
+        await makeView({
+            resModel: "account.move",
+            resId: 1,
+            serverData,
+            arch: fromArch,
+            type: "form",
+        });
+        let x2mbuttons = target.querySelector(".o_field_x2many_buttons");
+        assert.containsOnce(target, ".o_field_x2many_buttons", "should have rendered a x2many_buttons field");
+        assert.strictEqual(x2mbuttons.querySelectorAll("button").length, 2, "buttons should be rendered");
+    });
+
+    QUnit.test("component rendering: exactly 3 records on field", async function (assert) {
+        assert.expect(2);
+
+        await makeView({
+            resModel: "account.move",
+            resId: 4,
+            serverData,
+            arch: fromArch,
+            type: "form",
+        });
+        let x2mbuttons = target.querySelector(".o_field_x2many_buttons");
+        assert.containsOnce(target, ".o_field_x2many_buttons", "should have rendered a x2many_buttons field");
+        assert.strictEqual(x2mbuttons.querySelectorAll("button").length, 3, "buttons should be rendered");
+    });
+
+    QUnit.test("component rendering: more than 3 records on field", async function (assert) {
+        assert.expect(3);
+
+        await makeView({
+            resModel: "account.move",
+            resId: 6,
+            serverData,
+            arch: fromArch,
+            type: "form",
+        });
+        let x2mbuttons = target.querySelector(".o_field_x2many_buttons");
+        assert.containsOnce(target, ".o_field_x2many_buttons", "should have rendered a x2many_buttons field");
+        assert.strictEqual(x2mbuttons.querySelectorAll("button").length, 4, "buttons should be rendered");
+        assert.strictEqual(x2mbuttons.querySelectorAll("button")[3].innerText, "... (View all)", "button should be rendered");
+    });
+    QUnit.test("edit record and check if edits get discarded when click on one of the buttons and redirects to proper record", async function (assert) {
+        assert.expect(3);
+        const form = await makeView({
+            resModel: "account.move",
+            resId: 1,
+            serverData,
+            arch: fromArch,
+            type: "form",
+        });
+        patchWithCleanup(form.env.services.action, {
+            doAction(action) {
+                assert.deepEqual(action, {
+                    res_model: "account.move",
+                    res_id: 2,
+                    type: "ir.actions.act_window",
+                    views: [[false, "form"]],
+                });
+            }
```

Comment:
Not sure but shouldn't `doAction` call a super or something after the patch? (no clue)

Reply:
im not sure either, but I don't think it's necessairy, since when I debug the actoin on the component gets called too